packages/swap/src/pricing/index.ts

```diff
@@ -0,0 +1,84 @@
+import { request } from 'graphql-request';
+import { Asset } from '@/shared/enums';
+import env from '../config/env';
+import { gql } from '../gql/generated';
+import { CacheMap } from '../utils/dataStructures';
+import logger from '../utils/logger';
+import { deferredPromise } from '../utils/promise';
+
+type ChainAndAddress = {
+  chainId: string;
+  address: string;
+};
+
+// TODO: refactor to use internal asset
+export const chainflipAssetTokens: Record<Asset, ChainAndAddress> = {
+  FLIP: {
+    chainId: 'evm-1',
+    address: '0x826180541412D574cf1336d22c0C0a287822678A',
+  },
+  USDC: {
+    chainId: 'evm-1',
+    address: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+  },
+  DOT: {
+    chainId: 'dot',
+    address: '0x0000000000000000000000000000000000000000',
+  },
+  ETH: {
+    chainId: 'evm-1',
+    address: '0x0000000000000000000000000000000000000000',
+  },
+  BTC: {
+    chainId: 'btc',
+    address: '0x0000000000000000000000000000000000000000',
+  },
+};
+
+const priceCache = new CacheMap<string, Promise<number | undefined>>(10_000);
+
+export const GET_TOKEN_PRICE = gql(/* GraphQL */ `
```

Comment:
i am not sure if we should load the price from our (closed source) cache here bc we want people to be able to run the sdk without depending on our services. i think loading it from coingecko directly would be more appropriate